Add navController parameter to constructor

khalp opened the pull request:
First draft of navController updates

Note: doesn't include README updates yet, will do this after API has been finalized

I updated the sample to test out the custom navController/navigateToPane1 methods but we don't have to keep it that way:

https://user-images.githubusercontent.com/33138268/157948107-b8329c2c-57c8-40da-806c-29f334c89dd0.mp4

joyl1216 commented:
I am not sure whether we should let the developer pass in `navigateToPane1Handler` and `navigateToPane2Handler`. The two handlers should handle the basic navigation between two panes, as per the naming. Can we do something simpler, like this?
```
@Composable
fun TwoPaneLayout(
    modifier: Modifier = Modifier,
    paneMode: TwoPaneMode = TwoPaneMode.TwoPane,
    navController: NavHostController = rememberNavController(),
    pane1: @Composable TwoPaneScope.() -> Unit,
    pane2: @Composable TwoPaneScope.() -> Unit
)
```
If the developer would like to build some customized navigation, they can pass in their own navigationController, keep the basic `navigateToPane1` and `navigateToPane2` functions or implement their own navigation outside the TwoPaneLayout.

khalp commented:
> If the developer would like to build some customized navigation, they can pass in their own navigationController, keep the basic `navigateToPane1` and `navigateToPane2` functions or implement their own navigation outside the TwoPaneLayout.

I updated the constructor(s) to be simpler! I removed the handlers and default value for the navController from the new constructor (because in that case they should just use the original constructor, which calls `rememberNavController()`), and I also exposed the `currentSinglePane` property and `Screen` class. I thought both of those could be useful when implementing custom navigation methods (the `Screen` class has the values for the routes of panes 1 & 2).

Let me know if you think that works better!

joyl1216 commented on TwoPaneLayout/library/src/main/java/com/microsoft/device/dualscreen/twopanelayout/TwoPaneLayout.kt:
```diff
@@ -16,9 +16,11 @@ import androidx.compose.ui.layout.Layout
 import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
+import androidx.navigation.get
```
I don't think it is necessary.

joyl1216 commented on TwoPaneLayout/library/src/main/java/com/microsoft/device/dualscreen/twopanelayout/TwoPaneLayout.kt:
```diff
@@ -47,13 +49,17 @@ enum class TwoPaneMode {
 fun TwoPaneLayout(
     modifier: Modifier = Modifier,
     paneMode: TwoPaneMode = TwoPaneMode.TwoPane,
+    navController: NavHostController,
```
Can we have `navController: NavHostController = rememberNavController()`, instead of having the following 
```
@Composable
fun TwoPaneLayout(
    modifier: Modifier = Modifier,
    paneMode: TwoPaneMode = TwoPaneMode.TwoPane,
    pane1: @Composable TwoPaneScope.() -> Unit,
    pane2: @Composable TwoPaneScope.() -> Unit
)```

khalp replied:
This is up to us - if we want to just use one constructor, then it will force devs to also include the `androidx.navigation:navigation-compose` import (to resolve the `NavHostController` parameter)

Is that ok? Or would you prefer to keep them separate?
